### PR TITLE
Adding support for Windows and Virtualenv

### PR DESCRIPTION
**Pull request by sidhujasminder**
Extending Functionality over Windows and python virtualenv

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,127 +1,170 @@
...
+        else:
+            if hasattr(sys, 'real_prefix'):
+                self.install_data = os.path.join(os.environ.get('VIRTUAL_ENV'),'jsnapy')
+            elif 'win' in sys.platform:
+                self.install_data = os.path.join(os.path.expanduser('~'),'jsnapy')
+            else:
+                self.install_data = '/etc/jsnapy'
+
+        dir_path = self.install_data
+        mode = 0o777
```
Comment (vnitinv) [inline setup.py]: we can just keep read and write permission.

File: lib/jnpr/jsnapy/__init__.py [hunk 2]
```diff
@@ -15,14 +15,22 @@ class DirStore:
...
-        p_locations = [os.environ['JSNAPY_HOME']]   
-    p_locations.extend([os.path.join(os.path.expanduser('~'),'.jsnapy'),'/etc/jsnapy'])
+        p_locations = [os.environ['JSNAPY_HOME']]
+
+    if hasattr(sys, 'real_prefix'):
+        p_locations.extend([os.path.join(os.path.expanduser('~'), '.jsnapy'),
+                            os.path.join(os.environ.get('VIRTUAL_ENV'), 'jsnapy')])
+    elif 'win' in sys.platform:
+        p_locations.extend(
+            [os.path.join(os.path.expanduser('~'), '.jsnapy'), os.path.join(os.path.expanduser('~'), 'jsnapy')])
```
Comment (vnitinv) [inline lib/jnpr/jsnapy/__init__.py]: there are many pep8 violations.

File: setup.py (hunk 1 above)
Comment (vnitinv) [inline setup.py]: we don't need execution permission for text/test files.

Comment (sidhujasminder) [inline setup.py]: This was already done in the original setup file. Let me know if sure to make this change.

**Pull request closed without merging**